extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -167,20 +166,50 @@ export class AmpFlyingCarpet extends AMP.BaseElement {
       this.element,
       this.children_
     );
-    listen(this.element, AmpEvents.BUILT, this.layoutBuiltChild_.bind(this));
+    this.observeNewChildren_();
     this.firstLayoutCompleted_ = true;
     return Promise.resolve();
   }
 
+  /**
+   * Makes sure we schedule layout for elements as they are added
+   * to the flying carpet.
+   * @private
+   */
+  observeNewChildren_() {
+    const observer = new MutationObserver(changes => {
+      for (let i = 0; i < changes.length; i++) {
+        const {addedNodes} = changes[i];
+        if (!addedNodes) {
+          continue;
+        }
+        for (let n = 0; n < addedNodes.length; n++) {
+          const node = addedNodes[n];
+          if (!node.signals) {
```

Comment:
Can we guarantee that the element is upgraded to a CE by now? I think we need to use `isAmpElement` to check instead, then wait for upgrade, then wait for built.

Reply:
I think that we can actually assume they have been upgraded yet.

Either you get the sync call from native CEs or your get an earlier MutationObserver that has done it.